gobblin-core/src/main/java/org/apache/gobblin/converter/avro/JsonElementConversionWithAvroSchemaFactory.java

```diff
@@ -210,4 +217,116 @@ public Schema schema() {
       return this.schema;
     }
   }
-}
+
+  /**
+   * A converter to convert Union type to avro
+   * Here it will try all the possible converters for one type, for example, to convert an int value, it will try all Number converters
```

Comment:
Long story short: "there's no guarantee on preserving precision from Json to Avro type as the exact type information is clear from JsonElement"